dispatcher.go

```diff
@@ -85,52 +105,80 @@ func (b *backendNF) readFromServer() {
 		} else {
 			if response.Msgtype == gClient.MsgType_INIT_MSG {
 				log.Printf("Init Response from Server %s server: %s", response.AmfId, response.VerboseMsg)
+				//update GnbId for each gnb connection
+				ran, _ := context.Sctplb_Self().RanFindByGnbIp(response.GnbIpAddr)
+				if ran != nil && response.GnbId != "" {
+					ran.SetRanId(response.GnbId)
+				}
+			} else if response.Msgtype == gClient.MsgType_REDIRECT_MSG {
+				//TODO
 			} else {
-				b.sctp.Write(response.Msg)
+				//fetch ran connection based on GnbId
+				if response.GnbId == "" {
+					log.Printf("Received null GnbId from backend NF")
+				} else {
+					ran, _ := context.Sctplb_Self().RanFindByGnbIp(response.GnbIpAddr)
+					if ran != nil {
+						ran.SetRanId(response.GnbId)
+						ran.Conn.Write(response.Msg)
+					} else {
+						log.Printf("Couldn't fetch sctp connection with GnbId: %v", response.GnbId)
```

Comment:
here we will add sending message to all servers so that they can get the UEs to idle for that gnb ?